lib/engine-addon.js

```diff
@@ -387,10 +391,76 @@ module.exports = {
         return mergeTrees([
           externalTree,
           engineJSTree,
-          engineCSSTree
+          compiledEngineCSSTree
         ].filter(Boolean), { overwrite: true });
       };
 
+      this.compileLazyEngineStyles = function compileLazyEngineStyles(vendorTree, externalTree) {
+        var vendorCSSTree = buildVendorCSSTree.call(this, vendorTree);
+        var engineStylesOutputDir = 'engines-dist/' + this.name + '/assets/';
+
+        // Get base styles tree.
+        var engineStylesTree = buildEngineStyleTree.call(this);
+
+        var primaryStyleTree = null;
+        if (engineStylesTree) {
+          var preprocessedEngineStylesTree;
+          // `_addonPreprocessTree` was added in Ember 2.12 to ensure
+          // addons had preprocessTree / postprocessTree invoked properly
+          // this emulates the normal behavior before and after that version
```

Comment:
I'd rather we not support both versions. I don't want to accumulate version-specific cruft in a 0.X addon.